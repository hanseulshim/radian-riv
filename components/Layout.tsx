import Head from 'next/head'
import Header from './Header'
import Footer from './Footer'

type Props = {
  children: React.ReactNode
}

export default function Layout({ children }: Props) {
  return (
    <div id="layout">
      <Head>
        <title>Radian RIV</title>
        <link
          rel="icon"
          type="image/x-icon"
          href={`${process.env.baseUrl}/favicon.ico`}
        />
      </Head>
      <Header />
      {children}
      <Footer />
    </div>
  )
}
