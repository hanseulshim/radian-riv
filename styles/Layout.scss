#layout {
  min-height: 100%;
  display: flex;
  flex-direction: column;
  min-width: calc(1100px + 310px);
  width: 100%;
  margin-left: auto;
  margin-right: auto;

  #main {
    flex: 1;
    display: flex;
    margin-top: -1px;

    .container {
      padding: 3em 2em;
      flex: 1;
    }

    #sidebar {
      width: 310px;
      min-width: 310px;
      background-color: var(--prussian-blue);
      color: var(--white);
      display: flex;
      flex-direction: column;
      justify-content: space-between;

      .side-bar-content {
        display: flex;
        flex-direction: column;

        .route-container {
          display: flex;
          flex-direction: column;
          max-height: 1000px;
          overflow: auto;
        }

        .go-back {
          display: flex;
          align-items: center;
          padding: 20px 50px 0 50px;
          cursor: pointer;
          font-size: 0.6em;
          img {
            width: 20px;
            margin-right: 5px;
          }
        }
        .route-link {
          width: 100%;
          padding: 22.5px 25px 22.5px 50px;
          font-weight: 700;
          position: relative;
          &:hover {
            span {
              border-bottom: 1px solid var(--white);
            }
          }
          .label-icon {
            position: absolute;
            top: 30px;
            left: 28px;
            width: 12px;
          }
          &.active:before {
            content: '';
            position: absolute;
            left: 0;
            bottom: 0;
            border-bottom: 2px solid var(--robins-egg);
            width: 200px;
          }
          &.active .label-icon {
            position: absolute;
            top: 64px;
            left: 200px;
            width: 12px;
          }
          &.route-1 {
            background: var(--elephant);
          }
          &.route-2 {
            background: var(--cello);
          }
          &.route-3 {
            background: var(--san-juan);
          }
        }
      }
      .sidebar-footer {
        font-size: 0.6em;
        padding: 22.5px 75px 22.5px 50px;
        display: flex;
        justify-content: space-between;
      }
    }
  }
}

#header {
  img {
    width: 150px;
  }
  .header-spacer {
    padding: 10px 50px 0em 50px;
    &.primary-background {
      padding: 1em;
      background: var(--prussian-blue);
      color: var(--white);
      text-align: center;
    }
  }
  .header-row {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-bottom: 10px;

    .profile-row {
      display: flex;
      align-items: center;
      .profile-menu {
        display: flex;
        align-items: center;
        position: relative;
        .menu-container {
          position: absolute;
          box-shadow: 0px 3px 8px rgba(0, 0, 0, 0.1);
          top: 30px;
          left: -105px;
          width: 150px;
          display: flex;
          flex-direction: column;
          background: white;
          margin: 0;
          padding: 0;
          list-style-type: none;
          border-top: 1px solid var(--robins-egg);
          border-radius: 0px 0px 16px 16px;
          z-index: 1;
          li {
            cursor: pointer;
            padding: 7px 1em;

            &:hover {
              background: var(--light-gray);
            }
          }
        }
        #profile-logo {
          width: 25px;
          height: 25px;
          margin-left: 1em;
          cursor: pointer;
        }
      }
    }
  }
}

#footer {
  max-width: calc(1100px + 310px);
  width: 100%;
  height: 157px;
  margin: auto;
  padding: 1em 0;
  font-size: 0.6em;
  line-height: 0.75em;
  color: #668092;
  display: flex;
  align-items: center;
  background-color: var(--white);
  .logo-container {
    width: 350px;
    min-width: 350px;
    display: flex;
    justify-content: center;
  }

  .content {
    display: flex;
    flex-direction: column;

    .footer-links {
      margin-bottom: 20px;
      a {
        color: var(--oracle);
        font-size: 1rem;
        border-right: 1px solid var(--oracle);
        margin-right: 10px;
        padding-right: 10px;
      }
      a:last-child {
        border-right: none;
      }
    }
  }
}

#breadcrumbs {
  margin-bottom: 1em;
  display: flex;

  a {
    color: var(--patina);
    text-decoration: underline;
  }
  span {
    margin: 0 10px;
  }
}
