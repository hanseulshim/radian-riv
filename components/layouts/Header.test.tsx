import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Layout from './Layout'

describe('Layout', () => {
  let component
  beforeAll(() => {
    component = shallow(
      <Layout>
        <div>test</div>
      </Layout>
    )
  })
  it('has a div with id', () => {
    expect(component.find('#layout')).toHaveLength(1)
  })
  it('should have title', () => {
    expect(component.find('title').text()).toEqual('Radian RIV')
  })
  it('should have link to favicon', () => {
    expect(component.find('link').prop('href')).toContain('favicon.ico')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
