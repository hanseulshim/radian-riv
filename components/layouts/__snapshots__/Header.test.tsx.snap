// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div
  id="layout"
>
  <Head>
    <title>
      Radian RIV
    </title>
    <link
      href="undefined/favicon.ico"
      rel="icon"
      type="image/x-icon"
    />
  </Head>
  <Header />
  <div>
    test
  </div>
  <Footer />
</div>
`;
